### PR TITLE
Enable Code Coverage for Paddle refactor code

### LINKED ISSUE #3298: Reopen Code Coverage
If someone adds a new feature, the code coverage rate becomes lower, simply reject this pull request and remind him/her to add tests.

### PR DESCRIPTION
**Pull request by gangliao**
fix #3298 

- [x] Add code coverage for PaddlePaddle refactorization code
- [x] generate  coveralls report
- [x] Add git commit into JSON to bind coveralls with GitHub

image

https://coveralls.io/github/PaddlePaddle/Paddle

### CODE REVIEW

File: go/pserver/optimizer.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 package pserver
 
 // #cgo CFLAGS: -I ../../
-// #cgo LDFLAGS: ${SRCDIR}/client/c/libpaddle_go_optimizer.a -lstdc++ -lm
+// #cgo LDFLAGS: /Users/liaogang/baidu/Paddle/go/pserver/client/c/libpaddle_go_optimizer.a -lstdc++ -lm -g -O0 -fprofile-arcs -ftest-coverage
```
Comment (gangliao) [inline go/pserver/optimizer.go]: please ignore this modification, because  `go/pserver/.gitignore` is added.

**Comment by gangliao** [comment]
image

File: go/pserver/optimizer.go (hunk 1 above)
Comment (reyoung) [inline go/pserver/optimizer.go]: Use `git rm PATH_TO_FILE` to remove it from git.

**Review by reyoung — APPROVED** [review]
LGTM, except `optimizer.go` should be removed from git.

File: go/pserver/optimizer.go.in [hunk 2]
```diff
@@ -15,7 +15,7 @@
 package pserver
 
 // #cgo CFLAGS: -I ../../
-// #cgo LDFLAGS: ${SRCDIR}/client/c/libpaddle_go_optimizer.a -lstdc++ -lm
```
Comment (helinwang) [inline go/pserver/optimizer.go.in]: This will break building from Go without cmake. Do we really need this?

Comment (gangliao) [inline go/pserver/optimizer.go.in]: Maybe we can find another way to suppress this.

**Pull request closed without merging**